sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/Average.scala

```diff
@@ -75,18 +80,17 @@ abstract class AverageBase
   )
 
   protected def getMergeExpressions = Seq(
-    /* sum = */ Add(sum.left, sum.right, useAnsiAdd),
+    /* sum = */ add(sum.left, sum.right),
     /* count = */ count.left + count.right
   )
 
   // If all input are nulls, count will be 0 and we will get null after the division.
   // We can't directly use `/` as it throws an exception under ansi mode.
   protected def getEvaluateExpression(queryContext: String) = child.dataType match {
     case _: DecimalType =>
-      DecimalPrecision.decimalAndDecimal()(
-        Divide(
-          CheckOverflowInSum(sum, sumDataType.asInstanceOf[DecimalType], !useAnsiAdd, queryContext),
-          count.cast(DecimalType.LongDecimal), failOnError = false)).cast(resultType)
+      Divide(
```

Comment:
shall we use a `DecimalDivide` here?

Reply:
I think we do not need a `DecimalDivide`, before it was wrapped by `DecimalPrecision.decimalAndDecimal()` which applied `PromotePrecision` and `CheckOverflow`. This pr makes `Divide` suuport the semantics of `PromotePrecision` and `CheckOverflow`, so they are semantically equivalent.

Reply:
The intention was to check overflow on the division result, which was not possible before. Now if we can have a `DecimalDivide`, we can remove `CheckOverflowInSum` as well.